python_modules/dagster-graphql/dagster_graphql/implementation/external.py

```diff
@@ -25,58 +25,39 @@
     from dagster_graphql.schema.util import HasContext
 
 
-def get_full_external_job_or_raise(
-    graphene_info: HasContext, selector: PipelineSelector
-) -> ExternalPipeline:
-    from ..schema.errors import GraphenePipelineNotFoundError
-
-    check.inst_param(graphene_info, "graphene_info", ResolveInfo)
-    check.inst_param(selector, "selector", PipelineSelector)
-
-    if not graphene_info.context.has_external_job(selector):
-        raise UserFacingGraphQLError(GraphenePipelineNotFoundError(selector=selector))
-
-    return graphene_info.context.get_full_external_job(selector)
-
-
 def get_external_pipeline_or_raise(
```

Comment:
I like the consolidation of `get_subset_external_pipeline` and `get_external_pipeline_or_raise`. But can we keep `get_full_external_job_or_raise` separate? I worry that placing this functionality behind the `ignore_subset` flag might be easily missed

I think we can also rename `get_external_pipeline_or_raise` -> `get_external_job_or_raise`

Reply:
OK I added it back and factored out the common logic into a private function.

I'm hesitant to rename at this time because the return type is `ExternalPipeline`. I'd like to do a more consistent and thorough rename of legacy stuff later but I think it could be confusing to do it piecewise right now.